js/coinsbit.js

```diff
@@ -0,0 +1,799 @@
+        const method = this.safeString (this.options, 'fetchOrderBookMethod', type);
+        if (method === 'depth/result') {
+            const response = await this.publicGetDepthResult (this.extend (request, params));
+            //     {
+            //         asks: [
+            //             [  "46809.06069986", "2.651"  ],
+            //             [  "46819.80796146", "2.6401" ],
+            //             ...
+            //         ],
+            //         bids: [
+            //             [  "46799.39054516",  "2.702" ],
+            //             [  "46801.25196599",  "0.4"   ],
+            //             ...
+            //         ]
+            //     }
+            const priceKey = 0;
+            const amountKey = 1;
+            orderbook = this.parseOrderBook (response, symbol, undefined, 'bids', 'asks', priceKey, amountKey);
+        } else if (method === 'book') {
+            request['offset'] = this.safeString (params, 'offset', 0);
+            // ATM, they need to call separately (unfortunately)
+            request['side'] = this.safeString (params, 'side', 'buy');
+            const requestBids = this.publicGetBook (this.extend (request, params));
+            request['side'] = this.safeString (params, 'side', 'sell');
+            const requestAsks = this.publicGetBook (this.extend (request, params));
+            const responses = await Promise.all ([requestBids, requestAsks]);
+            // FOR EACH (BUY/SELL) REQUEST, THE RESPONSE IS SIMILAR (JUST 'side' PROPERTY IS DIFFERENT BETWEEN THEM)
+            //     {
+            //         success: true,
+            //         message: "",
+            //         result: {
+            //             offset: 0,
+            //             limit: 2,
+            //             total: 334,
+            //             orders: [
+            //                 {
+            //                     id: 8620142523,
+            //                     left: "1.56772",
+            //                     market: "BTC_USDT",
+            //                     amount: "1.56772",
+            //                     type: "limit",
+            //                     price: "46752.63032535",
+            //                     timestamp: 1640031161.544,
+            //                     side: "buy",
+            //                     dealFee: "0",
+            //                     takerFee: "0",
+            //                     makerFee: "0",
+            //                     dealStock: "0",
+            //                     dealMoney: "0"
+            //                 },
+            //                 ...
+            //             ]
+            //         },
+            //         code: 200
+            //     }
+            const resultBids = this.safeValue (responses[0], 'result', {});
```

Comment:
we cannot index the response with [0], instead we should use `this.safeValue (response, 0)`